bin/p2-exec/main_darwin.go

```diff
@@ -0,0 +1,77 @@
+package main
+
+import (
+	// #include <stdlib.h>
+	// #include <unistd.h>
+	// #include <sys/resource.h>
+	// #include <sys/types.h>
+	// #include <sys/syslimits.h>
+	// #include <pwd.h>
+	// #include <grp.h>
+	"C"
+	"unsafe"
+
+	"github.com/square/p2/pkg/user"
+)
+
+func nolimit() error {
+	maxFDs := C.OPEN_MAX
+	_, err := C.setrlimit(C.RLIMIT_NOFILE, &C.struct_rlimit{C.rlim_t(maxFDs), C.rlim_t(maxFDs)})
+	if err != nil {
+		return err
+	}
+
+	unlimit := &C.struct_rlimit{
+		C.rlim_t(C.RLIM_INFINITY),
+		C.rlim_t(C.RLIM_INFINITY),
+	}
+	_, err = C.setrlimit(C.RLIMIT_CPU, unlimit)
+	if err != nil {
+		return err
+	}
+	_, err = C.setrlimit(C.RLIMIT_DATA, unlimit)
+	if err != nil {
+		return err
+	}
+	_, err = C.setrlimit(C.RLIMIT_FSIZE, unlimit)
+	if err != nil {
+		return err
+	}
+	_, err = C.setrlimit(C.RLIMIT_MEMLOCK, unlimit)
+	if err != nil {
+		return err
+	}
+	_, err = C.setrlimit(C.RLIMIT_NPROC, unlimit)
+	if err != nil {
+		return err
+	}
+	_, err = C.setrlimit(C.RLIMIT_RSS, unlimit)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func changeUser(username string) error {
+	uid, gid, err := user.IDs(username)
```

Comment:
We are probably going to want to pass users and groups separately in the future. An ask has come from infosec that we execute applications with files owned by p2 but the group id being the user. This makes writing new files at install time difficult and thus discourages tampering.